src/utilities/selection/Selection.ts

```diff
@@ -13,9 +13,11 @@ export class Selection implements ISelection {
   private _keyToIndexMap: { [key: string]: number };
   private _anchoredIndex: number;
   private _onSelectionChanged: () => void;
+  private _onItemSelected: (item?: any, index?: number, isSelected?: boolean) => void;
   private _hasChanged: boolean;
 
-  constructor(onSelectionChanged?: () => void, getKey?: (item: IObjectWithKey, index?: number) => string) {
+  constructor(onSelectionChanged?: () => void, getKey?: (item: IObjectWithKey, index?: number) => string,
+  onItemSelected?: (item?: any, index?: number, isSelected?: boolean) => void) {
```

Comment:
i'm a little confused; why is this needed when the first param is onSelectionChanged? isn't the onSelectionChanged callback enough?